api_client/python/timesketch_api_client/sigma.py

```diff
@@ -22,8 +22,172 @@
+    def author(self):
+        """Returns the rule author."""
+        return self.get_attribute("author")
+
+    @property
+    def date(self):
+        """Returns the rule date."""
+        return self.get_attribute("date")
+
+    @property
+    def modified(self):
+        """Returns the rule modified date."""
+        return self.get_attribute("modified")
+
+    @property
+    def logsource(self):
+        """Returns the rule logsource."""
+        return self.get_attribute("logsource")
+
+    @property
+    def detection(self):
+        """Returns the rule detection."""
+        return self.get_attribute("detection")
+
+    @property
+    def references(self):
+        """Returns the rule references."""
+        return self.get_attribute("references")
+
+    @property
+    def status(self):
+        """Returns the rule status."""
+        return self.get_attribute("status")
+
+    def set_value(self, key, value):
+        """Sets the value for a given key
+
+        Args:
+            key: key to set the value
+            value: value to set
+
+        """
+        self._attr_dict[key] = value
+
+    def _load_rule_dict(self, rule_dict):
+        """Load a dict into a rule"""
+        for key, value in rule_dict.items():
+            self.set_value(key, value)
+
+    def from_rule_uuid(self, rule_uuid):
+        """Get a SigmaRule object from a rule UUID.
+
+        Args:
+            rule_uuid: Id of the sigma rule.
+
+        """
+        self.resource_uri = f"sigmarule/{rule_uuid}"
+
+        self.lazyload_data(refresh_cache=True)
+        objects = self.data.get("objects")
+        if not objects:
+            error_msg = (
+                "Unable to parse Sigma rule {0:s} with given text".format(
+                    rule_uuid
+                )
+            )
+            logger.error(error_msg)
+            raise ValueError(error_msg)
+        rule_dict = objects[0]
+        for key, value in rule_dict.items():
+            self.set_value(key, value)
+
+    def from_text(self, rule_text):
+        """Obtain a parsed Sigma rule by providing text.
+
+        Args:
+            rule_text: Rule text to be parsed.
+
+        Raises:
+            ValueError: If rule could not be parsed.
+        """
+        self.resource_uri = "{0:s}/sigmarule/text/".format(self.api.api_root)
+        data = {"title": "Get_Sigma_by_text", "content": rule_text}
+        response = self.api.session.post(self.resource_uri, json=data)
+        response_dict = error.get_response_json(response, logger)
+
+        objects = response_dict.get("objects")
+        if not objects:
+            error_msg = "Sigma rule Parsing error with provided rule"
+            logger.warning(error_msg)
+            raise ValueError(error_msg)
+
+        rule_dict = objects[0]
+        for key, value in rule_dict.items():
+            self.set_value(key, value)
+
+
 class Sigma(resource.BaseResource):
-    """Timesketch sigma object.
+    """DEPRECATED Timesketch sigma object.
```

Comment:
Same here, add logger telling the user that the method is deprecated.

Reply:
done